c8/plan/core.py

```diff
@@ -0,0 +1,22 @@
+from c8.request import Request
+from c8.exceptions import C8ServerError
+
+
+def build_request(method, endpoint, data=None):
+    if method == 'GET' or method == 'DELETE':
+        request = Request(
+            method=method,
+            endpoint=endpoint,
+        )
+        return request
+    elif method == 'POST' or method == 'PATCH':
+        request = Request(
+            method=method,
+            endpoint=endpoint,
+            data=data
+        )
+        return request
```

Comment:
The "if and elif" is not needed here you can directly pass data=data for any method, It should work because the Request constructor itself checks whether data is None or not.